crates/primitives/src/transaction/signature.rs

```diff
@@ -62,6 +62,11 @@ impl Signature {
     /// Output the `v` of the signature depends on chain_id
     #[inline]
     pub fn v(&self, chain_id: Option<u64>) -> u64 {
+        #[cfg(feature = "optimism")]
+        if self.r == U256::ZERO && self.s == U256::ZERO {
+            return 0
+        }
```

Comment:
what does this stand for? is it another heuristic for "is deposit transactoin"? in that case let's doc it